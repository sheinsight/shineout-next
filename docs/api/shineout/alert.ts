const api = JSON.parse('[{"title":"Alert","properties":[{"name":"className","tag":{"cn":"自定义类名","en":"Custom class name","default":"","version":""},"required":false,"type":"string "},{"name":"style","tag":{"cn":"自定义样式","en":"Custom style","default":"","version":""},"required":false,"type":"CSSProperties "}],"cn":"","en":""}]');
export default api;
