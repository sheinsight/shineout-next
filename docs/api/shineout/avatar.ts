const api = JSON.parse('[{"title":"Avatar","properties":[{"name":"className","tag":{"cn":"自定义类名","en":"Custom class name","default":"","version":""},"required":false,"type":"string "},{"name":"style","tag":{"cn":"自定义样式","en":"Custom style","default":"","version":""},"required":false,"type":"CSSProperties "},{"name":"size","tag":{"cn":"不同尺寸","en":"There are three built-in size: small、default、large.","default":"\\\"default\\\"","version":""},"required":false,"type":"\\\"small\\\" | \\\"large\\\" | \\\"default\\\""},{"name":"alt","tag":{"cn":"图像无法显示时的替代文本","en":"The alternative text for the image","default":"","version":""},"required":false,"type":"string "},{"name":"src","tag":{"cn":"图片类头像的资源地址或者图片元素","en":"The address of the image or the image element of the image class avatar","default":"","version":""},"required":false,"type":"ReactNode"},{"name":"icon","tag":{"cn":"设置头像的自定义图标","en":"Set the custom icon of the avatar","default":"","version":""},"required":false,"type":"ReactNode"},{"name":"children","tag":{"cn":"指定头像的形状","en":"Specifies the shape of the avatar","default":"\\\"circle\\\"","version":""},"required":false,"type":"ReactNode"},{"name":"shape","tag":{"cn":"头像文本内容，会根据内容尺寸自适应计算缩放比例","en":"The text content of the avatar, the scaling ratio will be calculated according to the content size","default":"","version":""},"required":false,"type":"\\\"circle\\\" | \\\"square\\\" "},{"name":"srcSet","tag":{"cn":"设置图片类头像响应式资源地址","en":"Set the responsive resource address of the image class avatar","default":"","version":""},"required":false,"type":"string "},{"name":"gap","tag":{"cn":"字符类型距离左右两侧边界单位像素","en":"The distance between the left and right edges of the character type in pixels","default":"","version":""},"required":false,"type":"number "},{"name":"draggable","tag":{"cn":"图片是否允许拖动","en":"Whether the image is allowed to be dragged","default":"true","version":""},"required":false,"type":"boolean "},{"name":"crossOrigin","tag":{"cn":"CORS 属性设置","en":"CORS attribute setting","default":"","version":""},"required":false,"type":"\\\"anonymous\\\" | \\\"use-credentials\\\" "},{"name":"onError","tag":{"cn":"图片加载失败的事件，返回 false 会关闭组件默认的 fallback 行为","en":"The event that the image fails to load, returning false will close the component\\\"s default fallback behavior","default":"","version":""},"required":false,"type":"(() => boolean) "}],"cn":"","en":"","sort":"0"},{"title":"Avatar.Group","properties":[{"name":"className","tag":{"cn":"自定义类名","en":"Custom class name","default":"","version":""},"required":false,"type":"string "},{"name":"style","tag":{"cn":"自定义样式","en":"Custom style","default":"","version":""},"required":false,"type":"CSSProperties "},{"name":"size","tag":{"cn":"不同尺寸","en":"There are three built-in size: small、default、large.","default":"\\\"default\\\"","version":""},"required":false,"type":"\\\"small\\\" | \\\"large\\\" | \\\"default\\\""},{"name":"shape","tag":{"cn":"头像文本内容，会根据内容尺寸自适应计算缩放比例","en":"The text content of the avatar, the scaling ratio will be calculated according to the content size","default":"","version":""},"required":false,"type":"\\\"circle\\\" | \\\"square\\\" "},{"name":"max","tag":{"cn":"设置头像组展示头像的最大数量","en":"Set the maximum number of avatars displayed in the avatar group","default":"","version":""},"required":false,"type":"number "},{"name":"renderMax","tag":{"cn":"自定义渲染隐藏的头像节点，hideAvatarNodes 为隐藏的节点 dom 数组，max 为隐藏的节点数量","en":"Custom rendering of hidden avatar nodes, hideAvatarNodes is an array of hidden node dom, max is the number of hidden nodes","default":"","version":""},"required":false,"type":"((hideAvatarNodes: ReactNode[], max: number) => ReactNode) "},{"name":"children","tag":{"cn":"头像节点","en":"Avatar node","default":"","version":""},"required":true,"type":"ReactNode"},{"name":"popover","tag":{"cn":"用于展示隐藏头像的 Popover 相关配置，配置属性为 Popover 部分属性: position、zIndex、getPopupContainer、adjust","en":"The event that the image fails to load, returning false will close the component\\\"s default fallback behavior","default":"","version":""},"required":false,"type":"Pick<PopoverProps, \\\"position\\\" | \\\"zIndex\\\" | \\\"getPopupContainer\\\" | \\\"adjust\\\"> "}],"cn":"","en":"","sort":"0"}]');
export default api;
