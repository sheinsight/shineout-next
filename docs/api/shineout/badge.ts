const api = JSON.parse('[{"title":"Badge","properties":[{"name":"className","tag":{"cn":"自定义类名","en":"Custom class name","default":"","version":""},"required":false,"type":"string "},{"name":"style","tag":{"cn":"自定义样式","en":"Custom style","default":"","version":""},"required":false,"type":"CSSProperties "},{"name":"children","tag":{"cn":"内容","en":"Content","default":"","version":""},"required":false,"type":"ReactNode"},{"name":"color","tag":{"cn":"自定义徽标颜色","en":"Custom badge color","default":"","version":""},"required":false,"type":"string "},{"name":"count","tag":{"cn":"展示的数字，大于 overflowCount 时显示 ${overflowCount}+，为 0 时隐藏","en":"Number to show, show ${overflowCount}+ when it is greater than overflowCount, hide when it is 0","default":"","version":""},"required":false,"type":"ReactNode"},{"name":"dot","tag":{"cn":"小点模式，开启后不展示数字","en":"Dot mode","default":"false","version":""},"required":false,"type":"boolean "},{"name":"offset","tag":{"cn":"偏移量","en":"Offset of the badge","default":"","version":""},"required":false,"type":"[number, number] "},{"name":"overflowCount","tag":{"cn":"封顶数值","en":"Offset of the badge","default":"","version":""},"required":false,"type":"number "},{"name":"showZero","tag":{"cn":"当数值为 0 时，是否展示 Badge","en":"Whether to show Badge when the number is 0","default":"","version":""},"required":false,"type":"boolean "},{"name":"status","tag":{"cn":"徽标状态","en":"Whether to show Badge when the number is 0","default":"","version":""},"required":false,"type":"\\\"success\\\" | \\\"processing\\\" | \\\"default\\\" | \\\"error\\\" | \\\"warning\\\" "},{"name":"text","tag":{"cn":"状态点文本，仅在配置了 status 属性后生效","en":"Status point text, only valid after configuring the status property","default":"","version":""},"required":false,"type":"ReactNode"},{"name":"size","tag":{"cn":"尺寸，仅在非 dot 模式下生效，支持 small 和 default 两种尺寸","en":"Size, only valid in non-dot mode, supports two sizes: small and default","default":"","version":""},"required":false,"type":"\\\"default\\\" | \\\"small\\\" "}],"cn":"","en":"","sort":"0"}]');
export default api;
