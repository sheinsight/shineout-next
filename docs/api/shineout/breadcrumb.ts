const api = JSON.parse('[{"title":"Breadcrumb","properties":[{"name":"className","tag":{"cn":"自定义类名","en":"Custom class name","default":"","version":""},"required":false,"type":"string "},{"name":"style","tag":{"cn":"自定义样式","en":"Custom style","default":"","version":""},"required":false,"type":"CSSProperties "},{"name":"data","tag":{"cn":"面包屑对象数组,见 data","en":"The array of breadcrumb objects, see data","default":"[]","version":""},"required":false,"type":"any[]"},{"name":"separator","tag":{"cn":"面包屑分隔符,可以是字符串或自定义的元素","en":"A breadcrumb separator which can be strings or custom elements","default":"\\\"/\\\"","version":""},"required":false,"type":"ReactNode"},{"name":"keygen","tag":{"cn":"生成每一项key的辅助方法。为 true 时，以数据项本身作为key，相当于 (d => d)；为函数时，使用此函数返回值；为string时，使用这个string对应的数据值。如 \\\"id\\\"，相当于 (d => d.id)","en":"Key generator.When it is true, the data itself is used as the key equivalent to (d => d);When it is a function, use its return value;When it is a string，ues the value of the string.For example, \\\"id\\\" is the same thing as (d) => d.id.","default":"","version":""},"required":false,"type":"true | ObjectKey<Item> | (data: Item, index?: number ) => string | number"},{"name":"renderItem","tag":{"cn":"自定义渲染","en":"Custom render","default":"","version":""},"required":false,"type":"((value: Item) => ReactNode) "},{"name":"max","tag":{"cn":"最大显示个数","en":"Maximum number of displays","default":"","version":""},"required":false,"type":"number "},{"name":"maxHeight","tag":{"cn":"下拉框最大高度，超出可以内滚","en":"The maximum height of the drop-down box. It can be scrolled if it exceeds the height.","default":"50vh","version":""},"required":false,"type":"string | number "}],"cn":"","en":"","sort":"0"},{"title":"BreadcrumbData","properties":[{"name":"onClick","tag":{"cn":"点击事件","en":"The click event","default":"","version":""},"required":false,"type":"MouseEventHandler<HTMLAnchorElement> "},{"name":"title","tag":{"cn":"显示内容","en":"Displayed content","default":"","version":""},"required":false,"type":"ReactNode"},{"name":"url","tag":{"cn":"链接地址，onClick 属性二选一","en":"Link address","default":"","version":""},"required":false,"type":"string "},{"name":"icon","tag":{"cn":"自定义图标","en":"Custom icon","default":"","version":""},"required":false,"type":"ReactNode"}],"cn":"","en":"","sort":"0"}]');
export default api;
