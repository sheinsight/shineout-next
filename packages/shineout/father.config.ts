import { defineConfig } from 'father';
export default defineConfig({
  extends: '../../.fatherrc.ts',
});
