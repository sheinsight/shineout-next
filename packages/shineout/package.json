{
  "name": "shineout",
  "version": "3.0.0",
  "description": "A components library for React",
  "keywords": [],
  "license": "ISC",
  "author": "",
  "sideEffects": false,
  "main": "./dist/cjs/index.js",
  "module": "./dist/esm/index.js",
  "typings": "./dist/cjs/index.d.ts",
  "scripts": {
    "build": "father build",
    "build:watch": "father dev",
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "dependencies": {
    "@sheinx/base": "workspace:*",
    "@sheinx/hooks": "workspace:*",
    "@sheinx/shineout-style": "workspace:*"
  },
  "peerDependencies": {
    "classnames": ">=2",
    "immer": ">=10",
    "react": ">=16.8"
  },
  "devDependencies": {
    "jszip": "^3.10.1"
  }
}
