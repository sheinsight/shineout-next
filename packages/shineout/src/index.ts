export { default as Alert } from './alert';
export { default as Button } from './button';
export { default as Checkbox } from './checkbox';
export { default as DatePicker } from './date-picker';
export { default as Dropdown } from './dropdown';
export { default as EditableArea } from './editable-area';
export { default as Form } from './form';
export { default as Icon } from './icon';
export { default as Image } from './image';
export { default as Input } from './input';
export { default as Popover } from './popover';
export { default as Radio } from './radio';
export { default as Rule } from './rule';
export { default as Spin } from './spin';
export { default as Switch } from './switch';
export { default as Tag } from './tag';
export { default as Tabs } from './tabs';
export { default as Textarea } from './textarea';
export { default as Tooltip } from './tooltip';
export { setConfig } from '@sheinx/base';

export * as TYPE from './type';
