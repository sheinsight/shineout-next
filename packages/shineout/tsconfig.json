{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "declaration": true,
    "baseUrl": "./",
    "paths": {
      "shineout": ["./src/index"],
      "@sheinx/base": ["../base/src/index"],
      "@sheinx/hooks": ["../hooks/src/index"]
    }
  },
  "include": ["src"]
}
