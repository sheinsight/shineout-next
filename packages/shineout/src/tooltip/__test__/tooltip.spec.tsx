import { render, cleanup, screen, fireEvent, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import Tooltip from '..';
import { Button } from 'shineout'
import mountTest from '../../tests/mountTest';
import { classLengthTest } from '../../tests/structureTest';
import {
  StyleProps,
  attributesTest,
  classContentTest,
  classTest,
  createClassName,
  delay,
  displayTest,
  snapshotTest,
  styleTest,
  textContentTest,
} from '../../tests/utils';
import TooltipBase from '../__example__/example-01-base';
import TooltipClick from '../__example__/example-03-01-click';
import TooltipDisabled from '../__example__/example-04-disabled-inner';

const SO_PREFIX = 'tooltip';
const {
  wrapper: tooltipClassName,
  content: tooltipContentClassName,
  wrapperOpen: tooltipWrapperOpenClassName,
  target: tooltipTargetClassName,
} = createClassName(SO_PREFIX, ['wrapper', 'content', 'target'], ['wrapperOpen'])

const TooltipDemo = ({
  className,
  style,
  trigger,
  position,
  delay,
  animation,
  priorityDirection,
  disabled,
  persistent,
}: {
  className?: string;
  style?: StyleProps;
  trigger?: 'click' | 'hover' | 'focus';
  position?: any;
  delay?: number;
  animation?: boolean;
  priorityDirection?: 'auto' | 'vertical' | 'horizontal' | undefined;
  disabled?: boolean;
  persistent?: boolean;
}) => (
  <Tooltip
    tip='hello'
    className={className}
    style={style}
    trigger={trigger}
    position={position}
    delay={delay}
    animation={animation}
    priorityDirection={priorityDirection}
    disabledChild={disabled}
    persistent={persistent}
  >
    <span>demo</span>
  </Tooltip>
);

const defaultStyle = 'position: absolute; z-index: 1051; left: 0px; transform: translateX(-50%); top: 0px;'

beforeAll(() => {
  jest.useFakeTimers();
});
afterAll(() => {
  jest.runAllTimers();
});
afterEach(cleanup);
describe('Tooltip[Base]', () => {
  mountTest(<TooltipDemo />);
  displayTest(Tooltip as React.FC, 'ShineoutTooltip');
  snapshotTest(<TooltipBase />);
  snapshotTest(<TooltipClick />, 'about click');
  snapshotTest(<TooltipDisabled />, 'about disabled');
  test('should render default', async () => {
    const { container } = render(<TooltipDemo />);
    textContentTest(container, 'demo');
    fireEvent.mouseEnter(screen.getByText('demo'));
    await waitFor(async () => {
      await delay(200);
    })
    const wrapper = document.querySelector(tooltipClassName)!;
    classTest(wrapper, tooltipWrapperOpenClassName)
    attributesTest(wrapper, 'data-soui-position', 'bottom');
    styleTest(wrapper, defaultStyle);
    const content = wrapper.querySelector(tooltipContentClassName)!;
    textContentTest(content, 'hello');
  });
  test('should render when set className and style', async () => {
    render(<TooltipDemo className='demo' style={{ color: 'black' }} />);
    fireEvent.mouseEnter(screen.getByText('demo'));
    await waitFor(async () => {
      await delay(200);
    })

    const wrapper = document.querySelector(tooltipClassName)!;
    classContentTest(wrapper, 'demo');
    styleTest(wrapper.querySelector(tooltipContentClassName)!, 'color: black;');
  });
  test('should render when mouse', async () => {
    render(<TooltipDemo />);

    fireEvent.mouseEnter(screen.getByText('demo'));
    await waitFor(async () => {
      await delay(200);
    });
    const wrapper = document.querySelector(tooltipClassName)!;
    classContentTest(wrapper, tooltipWrapperOpenClassName);
    styleTest(
      wrapper,
      defaultStyle,
    );
    fireEvent.mouseLeave(screen.getByText('demo'));

    await waitFor(async () => {
      await delay(200);
      classContentTest(wrapper, tooltipWrapperOpenClassName, false);
    });
  });
  test('should render when set trigger is click', async () => {
    render(<TooltipDemo trigger='click' />);

    fireEvent.click(screen.getByText('demo'));
    await waitFor(async () => {
      await delay(200);
    });
    const wrapper = document.querySelector(tooltipClassName)!;
    classContentTest(wrapper, tooltipWrapperOpenClassName);
      styleTest(
        wrapper,
        defaultStyle,
      );
    fireEvent.click(screen.getByText('demo'));
    await waitFor(async () => {
      await delay(200);
      classContentTest(wrapper, tooltipWrapperOpenClassName, false);
    });
  });
  test('should render when set different position', async () => {
    const positions = ['left', 'top', 'bottom', 'right'];
    render(
      <div>
        <Tooltip tip='hello world' trigger='hover' position='left'>
          <Button type='primary'>left</Button>
        </Tooltip>
        <Tooltip tip='hello world' trigger='hover' position='top'>
          <Button type='primary'>top</Button>
        </Tooltip>
        <Tooltip tip='hello world' trigger='hover' position='bottom'>
          <Button type='primary'>bottom</Button>
        </Tooltip>
        <Tooltip tip='hello world' trigger='hover' position='right'>
          <Button type='primary'>right</Button>
        </Tooltip>
      </div>
    );
    positions.forEach((item) => {
      fireEvent.mouseEnter(screen.getByText(item));
    })
    await waitFor(async () => {
      await delay(200);
    });
    const wrappers = document.querySelectorAll(tooltipClassName)!;
    wrappers.forEach((item, index) => {
      attributesTest(item, 'data-soui-position', positions[index]);
    });
    fireEvent.mouseEnter(screen.getByText(positions[0]));
    await waitFor(async () => {
      await delay(200);
      styleTest(
        wrappers[0],
        'position: absolute; z-index: 1051; top: 0px; transform: translateY(-50%) translateX(-100%); left: 0px;',
      );
    });
    fireEvent.mouseEnter(screen.getByText(positions[1]));
    await waitFor(async () => {
      await delay(200);
      styleTest(
        wrappers[1],
        'position: absolute; z-index: 1051; left: 0px; transform: translateX(-50%)translateY(-100%); top: 0px; transform-origin: center bottom;',
      );
    });
    fireEvent.mouseEnter(screen.getByText(positions[2]));
    await waitFor(async () => {
      await delay(200);
      styleTest(
        wrappers[2],
        'position: absolute; z-index: 1051; left: 0px; transform: translateX(-50%); top: 0px;',
      );
    });
    fireEvent.mouseEnter(screen.getByText(positions[3]));
    await waitFor(async () => {
      await delay(200);
      styleTest(
        wrappers[3],
        'position: absolute; z-index: 1051; top: 0px; transform: translateY(-50%); left: 0px;',
      );
    });
  });
  test('should render when set other postion', async () => {
    render(<TooltipDemo position='left-bottom' />);
    fireEvent.mouseEnter(screen.getByText('demo'));
    await waitFor(async () => {
      await delay(200);
      styleTest(
        document.querySelector(tooltipClassName)!,
        'position: absolute; z-index: 1051; top: 0px; transform: translateY(-100%) translateX(-100%); left: 0px;',
      );
    });
  });
  test('should render when set disabledChild', async () => {
    const { container } = render(<TooltipDisabled />);
    fireEvent.mouseEnter(screen.getByText('Disabled'))
    await waitFor(async () => {
      await delay(200);
    });
    styleTest(container.querySelector(tooltipTargetClassName)!, 'cursor: not-allowed;');
    classContentTest(document.querySelector(tooltipClassName)!, tooltipWrapperOpenClassName);
    fireEvent.mouseLeave(container.querySelector('button')!);
    await waitFor(async () => {
      await delay(200);
      classContentTest(
        document.querySelector(tooltipClassName)!,
        tooltipWrapperOpenClassName,
        false,
      );
    });
  });
  test('should render when set delay', async () => {
    render(<TooltipDemo delay={500} />);
    fireEvent.mouseEnter(screen.getByText('demo'));
    await waitFor(async () => {
      await delay(200);
      classContentTest(
        document.querySelector(tooltipClassName)!,
        tooltipWrapperOpenClassName,
      );
    });
    await waitFor(async () => {
      await delay(300);
      classContentTest(document.querySelector(tooltipClassName)!, tooltipWrapperOpenClassName);
    });
    fireEvent.mouseLeave(screen.getByText('demo'));
    await waitFor(async () => {
      await delay(200);
      classContentTest(
        document.querySelector(tooltipClassName)!,
        tooltipWrapperOpenClassName,
        false,
      );
    });
  });
  // TODO: animation is lose
  test('should render when set animation is false', () => {
    render(<TooltipDemo animation={false} />);
  });
  test('should render when set priorityDirection', async () => {
    render(<TooltipDemo priorityDirection='vertical' position='left' />);
    fireEvent.mouseEnter(screen.getByText('demo'));
    await waitFor(async () => {
      await delay(200);
      styleTest(
        document.querySelector(tooltipClassName)!,
        'position: absolute; z-index: 1051; top: 0px; transform: translateY(-50%) translateX(-100%); left: 0px;',
      );
    });
  });
  // Dont have disabled
  test('should render when set disable', async () => {
    render(<TooltipDemo disabled />);
    fireEvent.mouseEnter(screen.getByText('demo'));
    await waitFor(async () => {
      await delay(200);
    });
    const wrapper = document.querySelector(tooltipClassName)!;
    classContentTest(wrapper, tooltipWrapperOpenClassName);
  });
  test('should render when tip is lose', () => {
    render(
      // @ts-ignore
      <Tooltip>
        <span>demo</span>
      </Tooltip>,
    );
    classLengthTest(document, tooltipClassName, 0);
  });
  test('should render when children is lose', () => {
    const spyError = jest.spyOn(console, 'error').mockImplementation(() => {});
    render(
      // @ts-ignore
      <Tooltip tip='hello'></Tooltip>,
    );
    expect(spyError).toHaveBeenCalled();
    classLengthTest(document, tooltipClassName, 0);
    spyError.mockRestore();
  });
  test('should render when set persistent', async () => {
    render(<TooltipDemo persistent />);
    fireEvent.mouseEnter(screen.getByText('demo'));
    await waitFor(async () => {
      await delay(200);
    });
    const wrapper = document.querySelector(tooltipClassName)! as HTMLDivElement;
    expect(wrapper.style.pointerEvents).toBe('initial');
  });
});
