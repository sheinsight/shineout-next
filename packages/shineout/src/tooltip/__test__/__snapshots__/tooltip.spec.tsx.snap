// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip[Base] should render correctly  1`] = `
<div>
  <noscript />
  <button
    class="c-vk6maf soui-button-primary cd0yd2j soui-button-button"
    dir="ltr"
    type="button"
  >
    <span>
      hover me
    </span>
  </button>
</div>
`;

exports[`Tooltip[Base] should render correctly about click 1`] = `
<div>
  <noscript />
  <button
    class="c-vk6maf soui-button-primary cd0yd2j soui-button-button"
    dir="ltr"
    type="button"
  >
    <span>
      click me
    </span>
  </button>
</div>
`;

exports[`Tooltip[Base] should render correctly about disabled 1`] = `
<div>
  <noscript />
  <span
    class="c-eyzes3 soui-tooltip-target"
    style="cursor: not-allowed;"
  >
    <button
      class="chcg5ej soui-button-secondary cd0yd2j soui-button-button c-v6n58r soui-button-disabled"
      dir="ltr"
      disabled=""
      style="pointer-events: none;"
      type="button"
    >
      <span>
        Disabled
      </span>
    </button>
  </span>
</div>
`;
